backend/app/usecase/changelog/changelog_test.go

```diff
@@ -0,0 +1,180 @@
+// +build !integration all
+
+package changelog
+
+import (
+	"testing"
+	"time"
+
+	"github.com/short-d/app/mdtest"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/keygen"
+	"github.com/short-d/short/app/usecase/repository"
+	"github.com/short-d/short/app/usecase/service"
+)
+
+func TestPersist_CreateChange(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name          string
+		changeLog     []entity.Change
+		change        entity.Change
+		availableKeys []service.Key
+		hasErr        bool
+	}{
+		{
+			name: "Create change successfully",
+			changeLog: []entity.Change{
+				{
+					ID:              "12345",
+					Title:           "Title 1",
+					SummaryMarkdown: "Summary 1",
+				},
+				{
+					ID:              "54321",
+					Title:           "Title 2",
+					SummaryMarkdown: "Summary 2",
+				},
+			},
+			change: entity.Change{
+				Title:           "Title 3",
+				SummaryMarkdown: "Summary 3",
+			},
+			availableKeys: []service.Key{"test"},
+			hasErr:        false,
+		}, {
+			name: "no available key",
+			changeLog: []entity.Change{
+				{
+					ID:              "12345",
+					Title:           "Title 1",
+					SummaryMarkdown: "Summary 1",
+				},
+				{
+					ID:              "54321",
+					Title:           "Title 2",
+					SummaryMarkdown: "Summary 2",
+				},
+			},
+			change: entity.Change{
+				Title:           "Title 3",
+				SummaryMarkdown: "Summary 3",
+			},
+			availableKeys: []service.Key{},
+			hasErr:        true,
+		}, {
+			name: "ID already exists",
+			changeLog: []entity.Change{
+				{
+					ID:              "12345",
+					Title:           "Title 1",
+					SummaryMarkdown: "Summary 1",
+				},
+				{
+					ID:              "54321",
+					Title:           "Title 2",
+					SummaryMarkdown: "Summary 2",
+				},
+			},
+			change: entity.Change{
+				Title:           "Title 3",
+				SummaryMarkdown: "Summary 3",
+			},
+			availableKeys: []service.Key{"12345"},
+			hasErr:        true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			changeLogRepo := repository.NewChangeLogFake(testCase.changeLog)
+			keyFetcher := service.NewKeyFetcherFake(testCase.availableKeys)
+			keyGen, err := keygen.NewKeyGenerator(2, &keyFetcher)
+
+			mdtest.Equal(t, nil, err)
+
+			fakeTimer := mdtest.NewTimerFake(time.Now())
```

Comment:
Well, how do you verify time if we don't know what now is? The goal of using fake timer is control what 'now' is with preset data.